### PR TITLE
SDIT-2050: ✨ Try to get npm dependency action to upload sarif even if success

### PR DESCRIPTION
**Pull request by petergphillips**
- **SDIT-2050: ✨ Try to get npm dependency action to upload sarif even if success**
- **SDIT-2050: ✨ Bump to 0.6 for release**

### CODE REVIEW

File: .github/actions/security_npm_dependency/action.yml [hunk 1]
```diff
@@ -1,88 +0,0 @@
-name: 'Create and upload npm dependency reports'
```
Comment (petergphillips) [inline .github/actions/security_npm_dependency/action.yml:1]: not used anyway

File: .github/workflows/security_npm_dependency.yml [hunk 2]
```diff
@@ -25,70 +26,63 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       security-events: write
-    env:
-      parent_directory: ${{ github.event.repository.name == '' && '.' || github.event.repository.name}}
```
Comment (petergphillips) [inline .github/workflows/security_npm_dependency.yml:29]: not used

File: .github/workflows/security_npm_dependency.yml [hunk 3]
```diff
@@ -25,70 +26,63 @@ jobs:
...
     steps:
     - uses: actions/checkout@v4
     - uses: actions/setup-node@v4
       with:
         node-version: ${{ inputs.node_version }}
-    - id: npm-version
```
Comment (petergphillips) [inline .github/workflows/security_npm_dependency.yml:35]: switch `id` and `name` round

File: .github/workflows/security_npm_dependency.yml [hunk 4]
```diff
@@ -25,70 +26,63 @@ jobs:
...
+      shell: bash
+      run: python .github/scripts/auditjson_to_sarif.py npm-security-check-reports.json
+    - uses: github/codeql-action/upload-sarif@v3
+      id: npm-upload-sarif
+      with:
+        sarif_file: 'npm-security-check-reports.sarif'
+        category: npm-dependency-check
     - name: Audit for vulnerabilities - better
       id: npm-better
+      if: steps.npm.outcome == 'failure'
```
Comment (petergphillips) [inline .github/workflows/security_npm_dependency.yml:58]: only run second level audit if first fails

File: .github/workflows/security_npm_outdated.yml [hunk 5]
```diff
@@ -52,12 +52,12 @@ jobs:
...
       run: npm outdated ${{ inputs.dependencies_to_check }} > npm-outdated-reports.txt
       continue-on-error: true
-    - uses: ministryofjustice/hmpps-github-actions/.github/actions/security_npm_outdated@v0.5 # WORKFLOW_VERSION
+    - uses: ministryofjustice/hmpps-github-actions/.github/actions/security_npm_outdated@v0.6 # WORKFLOW_VERSION
       if: steps.npm.outcome == 'failure'
       with:
         channel_id: ${{ inputs.channel_id }}
         SLACK_BOT_TOKEN: ${{ secrets.HMPPS_SRE_SLACK_BOT_TOKEN }}
-    - uses: slackapi/slack-github-action@e28cf165c92ffef168d23c5c9000cffc8a25e117 # v1.24.0
+    - uses: slackapi/slack-github-action@37ebaef184d7626c5f204ab8d3baff4262dd30f0 # v1.27.0
```
Comment (petergphillips) [inline .github/workflows/security_npm_outdated.yml:60]: upgrade all to latest slack version

**Review by james-jdgtl — APPROVED** [review]
Looks good!

**Pull request merged**